Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -92,44 +94,51 @@ public NSIndexPath GetIndexForItem(object item)
 			}
 		}
 
-		void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
+		async void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 		{
 			if (Device.IsInvokeRequired)
 			{
-				Device.BeginInvokeOnMainThread(() => CollectionChanged(args));
+				Device.BeginInvokeOnMainThread(async () => await CollectionChanged(args));
```

Comment:
maybe use `Device.InvokeOnMainThreadAsync` so it can just get awaited inside this method?

Should this method be try catched since it's async/void?

Reply:
What does a try/catch buy us here?